projects/archiver/src/indexer/getIssueSummary.ts

```diff
@@ -0,0 +1,88 @@
+import { fromPairs } from 'ramda'
+import { oc } from 'ts-optchain'
+import {
+    IssuePublication,
+    IssueSummary,
+    ImageSize,
+    imageSizes,
+    notNull,
+} from '../../../common/src'
+import { getPublishedId } from '../publishedId'
+import { Bucket, s3 } from '../s3'
+import { issue } from '../../main'
+
+export const getIssueSummary = async (
+    issuePublication: IssuePublication,
+): Promise<IssueSummary | undefined> => {
+    const { edition, issueDate } = issuePublication
+
+    const publishedId = getPublishedId(issuePublication)
+
+    const assetKeyList = await s3
+        .listObjectsV2({
+            Bucket,
+            Prefix: `zips/${publishedId}`,
+        })
+        .promise()
+
+    const assetKeys = oc(assetKeyList)
```

Comment:
can we have some unit tests for some of those functions ? i think for `getIssueSummary` you can extract the function that receives `assetKeyList` as input without the need to call s3 client and do all that logic.

It will be easier to understand the intention of the code especially for newbies like me :)